### ISSUE TITLE
React-Router docs contribution required

### ISSUE THREAD

**Issue by AdityaSrivastava185** [AdityaSrivastava185/Reactjs-For-Beginners#32]
### Is your feature request related to a problem? Please describe.

I often find that new users struggle to understand the nuances of nested routes in React Router. Specifically, the current documentation does not provide enough real-world examples or clear explanations of how to effectively implement nested routing in various scenarios. This can lead to confusion and incorrect implementations, which ultimately affects user experience.

### Describe the solution you'd like

I propose enhancing the React Router documentation by including detailed examples of nested routes. These examples should cover common use cases, such as dynamic routes with parameters, route-based code splitting, and protecting routes with authentication. Additionally, a step-by-step guide or video tutorial could greatly benefit users who are visual learners.

### Describe alternatives you've considered

While there are some third-party tutorials and articles available online, they often lack the authority and up-to-date information that comes from official documentation. Additionally, community forums can provide some insights, but they are not always comprehensive or reliable.

### Additional context

Having clear and thorough documentation will not only help new users adopt React Router more easily but also improve overall community support by reducing the number of questions related to nested routing. Screenshots and code snippets could further illustrate the concepts effectively.

**Comment by Nitheesh-2003** [comment]
Hi there,
Can you please assign  this to me ?

**Comment by AdityaSrivastava185** [comment]
Sure  , happy contributions 😊

**Comment by Nitheesh-2003** [comment]
Hi @AdityaSrivastava185 , I have a small doubt there is already a react-router  documentation in the src.

**Comment by AdityaSrivastava185** [comment]
Yes @Nitheesh-2003 , there are lot of things which needs to be improved , keep the things simple , easy to understand and structured for learning

**Comment by Nitheesh-2003** [comment]
Sure @AdityaSrivastava185, Will work on it

**Comment by Nitheesh-2003** [comment]
Hi @AdityaSrivastava185 

I have updated the documentations of React Router.
Can you please review it and merge it.

Thank you
